pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -524,10 +524,12 @@ func (os *OpenStack) GetDevicePath(volumeID string) string {
 
 	if devicePath == "" {
 		devicePath = os.GetDevicePathFromInstanceMetadata(volumeID)
-	}
-
-	if devicePath == "" {
-		glog.Warningf("Failed to find device for the volumeID: %q", volumeID)
+		if devicePath == "" {
+			glog.Warningf("Failed to find device for the volumeID: %q", volumeID)
+			return ""
```

Comment:
i think the original is clearer in intent. 

for the new changes, if you delete line 529,530,531 then it's probably clearer (personally)